### PR TITLE
Disallow binaries by default on non-glibc Linux

### PR DESCRIPTION
**Pull request by ararslan**
Replaces #327

### CODE REVIEW

**Comment by tkelman** [comment]
There should really be a way of detecting or annotating an actual minimum here for any given binary.

**Comment by ararslan** [comment]
That would be neat but that sounds like a pretty significant change, since you'd also have to inspect all downloaded binaries and reliably determine the glibc version they require. Worth thinking about, though that's more than I wanted to get into with this PR.

File: src/dependencies.jl [hunk 1]
```diff
@@ -722,14 +722,16 @@ function check_system_handle!(ret,dep,handle)
 end
 
 # Default installation method
-if Compat.Sys.isapple()
-    defaults = [Binaries,PackageManager,SystemPaths,BuildProcess]
-elseif Compat.Sys.islinux() || (Compat.Sys.isbsd() && !Compat.Sys.isapple())
-    defaults = [PackageManager,SystemPaths,Binaries,BuildProcess]
+defaults = if Compat.Sys.isbsd()
+    [Binaries, PackageManager, SystemPaths, BuildProcess]
```
Comment (tkelman) [inline src/dependencies.jl]: I don't think Binaries are safe to enable for non-Apple bsd here.

Comment (ararslan) [reply src/dependencies.jl]: Fixed in https://github.com/JuliaLang/BinDeps.jl/pull/334

**Pull request merged**